hudi-hadoop-common/src/main/java/org/apache/hudi/common/util/AvroOrcUtils.java

```diff
@@ -514,15 +514,20 @@ public static Object readFromVector(TypeDescription type, ColumnVector colVector
         }
       case STRING:
         String stringType = avroSchema.getProp(GenericData.STRING_PROP);
+        Object parsedValue;
         if (stringType == null || !stringType.equals(StringType.String)) {
           int stringLength = ((BytesColumnVector) colVector).length[vectorPos];
           int stringOffset = ((BytesColumnVector) colVector).start[vectorPos];
           byte[] stringBytes = new byte[stringLength];
           System.arraycopy(((BytesColumnVector) colVector).vector[vectorPos], stringOffset, stringBytes, 0, stringLength);
-          return new Utf8(stringBytes);
+          parsedValue = new Utf8(stringBytes);
         } else {
-          return ((BytesColumnVector) colVector).toString(vectorPos);
+          parsedValue = ((BytesColumnVector) colVector).toString(vectorPos);
         }
+        if (avroSchema.getType() == Schema.Type.ENUM && !parsedValue.toString().isEmpty()) {
+          return new GenericData.EnumSymbol(avroSchema, parsedValue.toString());
```

Comment:
`parsedValue.toString()` called 2 times here.

Reply:
We support it as a string in the file, similar to parquet. In our parquet reading we translate to enum when reading as avro so this is for consistency across base file formats.